rfcs/0008-leaderless-dns.md

```diff
@@ -0,0 +1,221 @@
+
+Where a clusterID is not specified the DNS Operator will act with the DNS Record as the authority and ensure the entire host is updated in the zone to exactly match the specified DNS Record when cleaning up in this scenario, the host is fully removed from the DNS Provider zone.
+
+### API Updates
+New configuration options in the providerSecret:
+- `ClusterID`
+  - When absent the DNS Operator will consider itself the owner of the zone and the local DNS Records as the authority.
+  - When present the DNS Operator will consider the zone the authority and only insert/update any missing records from the local DNS Records.
+  - This can only be alphanumeric and have no whitespace, and is case-insensitive
+  - Used when generating an A record, e.g. `<hash>-<clusterID>.<lb-id>.<host...>`
+- `RefreshInterval`
+  - When absent it defaults to the shortest TTL of created records. This is how often in seconds to reconcile the DNS Policy in order to pull fresh zone values from the DNS Provider.
+
+New fields added to the DNS Policy CRD: 
+- `HealthCheck.MultiClusterFailureThreshold`
+  - When absent defaults to `1`, what percentage of clusters need to be reporting an endpoint unhealthy on a remote IP for it to be removed from the zone by a cluster that does not own that IP.
+
+New fields added to the DNS Record CRD:
+- `HealthCheck`
+  - Propagated from the DNS Policy, to facilitate reconciling health checks in the DNS controller.
+- `ProviderSecret`
+  - Propagated from the DNS Policy to facilitate pulling the zone in the DNS controller.
+
+### Changes to DNS Policy reconciler
+
+This will primarily need to be made aware of the new fields on the DNS Policy (Healthcheck block and providerSecret field) that it will need to propagate into any DNS Records that it creates or updates. Otherwise the logic of this controller remains largely the same.
+
+### DNS Operator
+
+A new component that will deal with taking the DNS Records output by the DNSPolicy reconciler and ensure they are accurately reflected in the DNS Provider zone.
+
+This component should already exist, but in a fashion that assumes the DNS Record as the source of truth and always ensures that the zone matches the local DNS Record CR.
+
+There are several changes required in this component:
+
+#### Build DNS Records from the provider zone
+
+When the DNS Operator reconciles a DNS Record (kuadrantRecord) it will first look to the referenced providerSecret to see if a ClusterID is set, if so the operator will first pull the most recent records from the DNS Provider and construct a separate DNS Record which accurately reflects the state of the zone in the provider (zoneRecord).
```

Comment:
kuadrant record is a bit confusing, as above

Reply:
I'm just trying to differentiate between the DNS Record created by Kuadrant and the DNS Record created by the DNS Operator, they are still both instances of the same DNSRecord CRD.